apps/omg_eth/lib/eth/root_chain.ex

```diff
@@ -236,24 +234,38 @@ defmodule OMG.Eth.RootChain do
   end
 
   def deposit_blknum_from_receipt(receipt) do
-    [{_, deposit_blknum, _, _}] = filter_receipt_events(receipt["logs"], "Deposit(address,uint256,address,uint256)")
+    [%{blknum: deposit_blknum}] =
+      filter_receipt_events(receipt["logs"], "Deposit(address,uint256,address,uint256)", &decode_deposit/1)
+
     deposit_blknum
   end
 
-  @spec filter_receipt_events([%{topics: [binary], data: binary()}], binary) :: [tuple]
-  def filter_receipt_events(receipt_logs, signature) do
+  defp decode_deposit(log) do
+    indexed_keys = [:owner, :blknum]
+    indexed_keys_types = [:address, {:uint, 256}]
+    non_indexed_signature = "Deposit(address,uint256)"
```

Comment:
I think skipping signature and not using parse_event/2 would be a better, less confusing way of dealing with this. I.e. there is no such thing as "Deposit(address,uint256)". Admitting that parse_event/2 is limited and should be rewritten later is probably a more readable solution.